pkg/storegateway/bucket_stores.go

```diff
@@ -152,10 +153,31 @@ func (u *BucketStores) InitialSync(ctx context.Context) error {
 // SyncBlocks synchronizes the stores state with the Bucket store for every user.
 func (u *BucketStores) SyncBlocks(ctx context.Context) error {
 	return u.syncUsersBlocks(ctx, func(ctx context.Context, s *store.BucketStore) error {
-		return s.SyncBlocks(ctx)
+		return u.syncUsersBlocksWithRetries(ctx, s)
```

Comment:
I'm not sure about this design. We pass our own function to `u.syncUsersBlocks()`. I was expecting something like this:

```
return u. syncUsersBlocksWithRetries(ctx, func(ctx context.Context, s *store.BucketStore) error {
  return s.SyncBlocks(ctx)
}
```

And then having `syncUsersBlocksWithRetries()` calling `syncUsersBlocks()`.

Am I missing anything?

Reply:
Nope. I'll get on it!